### DISCUSSION TITLE
RSA Setup

### DISCUSSION THREAD

**Discussion by jdpayne1969** [strongswan/strongswan#290] [Q&A]
Hi,
New at this and I know you all are busy, but I could use more detailed instructions please.  Seems like it should be a simple setup, but I must be missing some step.

I've been following:
https://wiki.strongswan.org/projects/strongswan/wiki/SecurityRecommendations

The Authentication with RSA and ECDSA keys section with the steps.  It would be great to see more detail in the steps.  Like on machine A:
ipsec pki --gen -s 4096 --outform pem > A.key
.
.
.
As it is not clear where A.key, A.pub, etc go on each machine.  Been trying for days and consistently get:
16[IKE] no private key found for '<IPADDRESS>'

Thanks in advance,
Jeff

**Comment by Thermi** [comment]
Public keys go onto the other hosts.
Private keys stay on the host that generated it. For ipsec.conf and ipsec.secrets, the private key needs to be referenced in ipsec.secrets.
For swanctl.conf, only the public key needs to be referenced. the swanctl binary finds the private key to it by itself, unless a password is needed to decrypt it.

To be in any way efficient in finding the reason for why it's not working, you need to configure debug logging (#196) and read the log. The configs in #196 have a reasonable amount of granularity. Any higher and you get too many binary dumps.

**Reply by jdpayne1969** [reply]
Thanks.  That what I've always thought.  I'll check my ipec. files.

Is there any easy way to check the public key to be sure it is correctly set, for example the correct CN?  I've read in other discussions that sometimes these get criss crossed.

I'll do the debug logging on Monday.

Thanks,
Jeff

**Reply by Thermi** [reply]
raw public keys don't have any metadata fields like CN, issuer, valid until, ... .
They're just the public key.
The ID is derived from the public key in a deterministic fashion. I believe it's the public key in hex form or something. You can see it though because it defaults to it.

**Reply by Thermi** [reply]
Did the issue get fixed?
